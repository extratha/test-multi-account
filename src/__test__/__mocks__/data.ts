export const data = {
  test: 'test'
}