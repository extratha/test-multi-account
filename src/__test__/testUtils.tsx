import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
