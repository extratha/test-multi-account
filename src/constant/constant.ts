export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};
