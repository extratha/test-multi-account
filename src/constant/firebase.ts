export const remoteConfigKey = {
  SYMPTOM_CHECKER: "SymptomChecker",
  LAB_INTERPRET_REQUIRE_FIELDS: "LabInterpretRequireFields",
};
