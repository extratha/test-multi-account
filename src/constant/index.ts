export const ENV = {
  BASE_API_URL: process.env.NEXT_PUBLIC_BASE_API_URL || "MOCK_NEXT_PUBLIC_BASE_API_URL",
  FIREBASE_API_KEY: process.env.NEXT_PUBLIC_FIREBASE_API_KEY || "MOCK_NEXT_PUBLIC_FIREBASE_API_KEY",
  FIREBASE_AUTH_DOMAIN: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN || "MOCK_NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN",
  FIREBASE_PROJECT_ID: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_PROJECT_ID",
  FIREBASE_STORAGE_BUCKET:
    process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET || "MOCK_NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET",
  FIREBASE_MESSAGING_SENDER_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID",
  FIREBASE_APP_ID: process.env.NEXT_PUBLIC_FIREBASE_APP_ID || "MOCK_NEXT_PUBLIC_FIREBASE_APP_ID",
  FIREBASE_MEASUREMENT_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID",
};

export const SESSION = {
  ACCESS_TOKEN: "access-token",
  REFRESH_TOKEN: "refresh-token",
  RESET_PASSWORD_TOKEN: "reset-password-token",
  X_ROBOT_TOKEN: "x-robot-token",
};

export const MENU_CONDITION = {
  LOGOUT: "logout",
};

export const NAVIGATION = {
  NOT_FOUND: "/not-found",
  LOGIN: "/login",
  HOME: "/home",
  FORGET_PASSWORD: "/forget-password",
  SET_NEW_PASSWORD: "/set-new-password",
  CONSENT_TERMS_CONDITIONS: "/consents/terms-conditions",
  CONSENT_PRIVACY_POLICIES: "/consents/privacy-policies",
  AI_INTERPRET_RESULT: "/ai-interpret/interpret",
  AI_INTERPRET_TRY_EXAMPLE_DATA: "/ai-interpret/try-example-data",
  AI_INTERPRET_TRY_INPUT_DATA: "/ai-interpret/try-input-data",
  SYMPTOM_CHECKER: "/symptom-checker",
  SETTINGS_TERMS_AND_CONDITIONS: "/settings/terms-and-conditions",
  SETTINGS_PRIVACY_POLICY: "/settings/privacy-policy",
  PUBLIC_TERMS_AND_CONDITIONS: "/public/consents/terms-conditions",
  PUBLIC_PRIVACY_POLICY: "/public/consents/privacy-policies",
};

export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};

export const GROUP_NAME = {
  GENERAL_CHECK_UP: "generalCheckUp",
  HEMATOLOGY_BG: "hematologyBG",
  HEMATOLOGY_CBC: "hematologyCBC",
  LIPID_PROFILE: "lipidProfile",
  BLOOD_CHEMISTRY: "bloodChemistry",
  RENAL_FUNCTION_TEST: "renalFunctionTest",
  LIVER_FUNCTION_TEST: "liverFunctionTest",
  URINE_ANALYSIS: "urineAnalysis",
  STOOL_EXAMINATION: "stoolExamination",
  PAP_SMEAR: "papSmear",
  TUMOR_MARKER: "tumorMarker",
};

export const GENERAL_CHECK_UP = {
  AGE: "age",
  GENDER: "gender",
  BODY_WEIGHT: "body_weight",
  BODY_HEIGHT: "body_height",
  WAIST_LINE: "waist_line",
  BLOOD_PRESSURE_SYSTOLIC: "blood_pressure_systolic",
  BLOOD_PRESSURE_DIASTOLIC: "blood_pressure_diastolic",
  PULSE: "pulse",
  RESPIRATORY_RATE: "respiratory_rate",
  TEMPERATURE: "temperature",
};

export const HEMATOLOGY_BLOOD = {
  BLOOD_GROUP: "blood_group",
  BLOOD_GROUP_RH: "blood_group_rh",
};

export const HEMATOLOGY_BLOOD_GROUP = {
  HEMATOLOGY_BLOOD: "bloodGroup",
};

export const HEMATOLOGY_CBC = {
  HB: "hb_value",
  HCT: "hct_value",
  RBC_MORPHOLOGY: "rbc_morphology_value",
  MCV: "mcv_value",
  MCH: "mch_value",
  MCHC: "mchc_value",
  RDW: "rdw_value",
  WBC: "wbc_value",
  NEUTROPHIL: "neutrophil_value",
  LYMPHOCYTE: "lymphocyte_value",
  MONOCYTE: "monocyte_value",
  EOSINOPHIL: "eosinophil_value",
  BASOPHIL: "basophil_value",
  PLATELET_COUNT: "platelet_count_value",
};

export const HEMATOLOGY_CBC_GROUP = {
  HEMOGLOBIN: "hemoglobin",
  WHITE_BLOOD: "whiteBlood",
  RED_BLOOD: "redBlood",
};

export const INTERPRET_STATUS = {
  PENDING: "pending",
  SUCCESS: "success",
  FAILED: "failed",
};

export const ASR_SERVICE = {
  ORDER_ASR: "order-asr",
  SOAP_ASR: "soap-asr",
};

export const ASR_SERVICE_KEY: Record<string, string> = {
  [ASR_SERVICE.ORDER_ASR]: "orderAsr",
  [ASR_SERVICE.SOAP_ASR]: "soapAsr",
};

export const CONSENT_TYPE = {
  TERMS_CONDITIONS: "terms-conditions",
  PRIVACY_POLICIES: "privacy-policies",
};

export const SUBMIT_CONSENT_TYPE = {
  [CONSENT_TYPE.TERMS_CONDITIONS]: "terms",
  [CONSENT_TYPE.PRIVACY_POLICIES]: "privacy_policies",
};
