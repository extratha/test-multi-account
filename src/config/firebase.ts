// Import the functions you need from the SDKs you need
import { envConfig } from "@/constant/env";
import { initializeApp } from "firebase/app";

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: envConfig.firebase.apiKey,
  authDomain: envConfig.firebase.authDomain,
  projectId: envConfig.firebase.projectId,
  storageBucket: envConfig.firebase.storageBucket,
  messagingSenderId: envConfig.firebase.messagingSenderId,
  appId: envConfig.firebase.appId,
  measurementId: envConfig.firebase.measurementId,
};

const firebaseApp = initializeApp(firebaseConfig);
export default firebaseApp;
