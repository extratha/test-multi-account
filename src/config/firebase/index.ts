// Import the functions you need from the SDKs you need
import { initializeApp, getApps } from "firebase/app";
import { getAnalytics } from "firebase/analytics";
import { staticEnvConfig } from "@/constant/env";

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: staticEnvConfig.firebase.apiKey,
  authDomain: staticEnvConfig.firebase.authDomain,
  projectId: staticEnvConfig.firebase.projectId,
  storageBucket: staticEnvConfig.firebase.storageBucket,
  messagingSenderId: staticEnvConfig.firebase.messagingSenderId,
  appId: staticEnvConfig.firebase.appId,
  measurementId:staticEnvConfig.firebase.measurementId
};

// Initialize Firebase
export const app = getApps().length === 0 ? initializeApp(firebaseConfig) : getApps()[0];

export const analytics = (typeof window !== "undefined") ? getAnalytics(app) : undefined;