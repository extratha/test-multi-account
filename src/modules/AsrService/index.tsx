"use client";
import { styled } from "@mui/material";
import { useParams } from "next/navigation";
import { useEffect, useState } from "react";

import { Page } from "@/components/Page";
import DashboardPage from "@/components/Page/DashboardPage";
import { ASR_SERVICE_KEY } from "@/constant";
import useTranslation from "@/locales/useLocale";
import { usePageLoadingStore, useUserProfileStore } from "@/store";
import { AsrConfig } from "@/types/model.ui";
import { getDashboardMenuConfig } from "@/utils/firebase";

export interface AsrServiceParams {
  service: string;
  [key: string]: string | string[];
}

const IFrame = styled("iframe")({
  width: "100%",
  height: "100%",
  border: "0px",
});

const AsrService = () => {
  const { service } = useParams<AsrServiceParams>();
  const { setPageLoading } = usePageLoadingStore();
  const { data } = useUserProfileStore();
  const { translation } = useTranslation();

  const [config, setConfig] = useState<AsrConfig>();

  const getTitle = () => {
    return translation(`${ASR_SERVICE_KEY[service]}.title`);
  };

  const fetchConfig = async () => {
    try {
      setPageLoading(true);
      const { asr } = await getDashboardMenuConfig();
      const pageConfig = asr.find((item) => item.name === service);

      if (!pageConfig) throw new Error("config not found");

      setConfig(pageConfig);
      setPageLoading(false);
    } catch (error) {
      setPageLoading(false);
    }
  };

  useEffect(() => {
    fetchConfig();
  }, []);

  return (
    <>
      {config && (
        <Page title={getTitle()}>
          <DashboardPage>
            <IFrame src={`${config.url}?source=${data.corporate}`} allow="camera; microphone" />
          </DashboardPage>
        </Page>
      )}
    </>
  );
};

export default AsrService;
