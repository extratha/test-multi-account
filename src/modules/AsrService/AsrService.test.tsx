import * as RemoteConfig from "firebase/remote-config";

import { mockDashboardMenuConfigResult } from "@/__mocks__/data";
import { ASR_SERVICE } from "@/constant";
import * as UsePageLoadingStore from "@/store/usePageLoadingStore";
import { PageLoadingStore } from "@/store/usePageLoadingStore";
import * as UseUserProfileStore from "@/store/useUserProfileStore";
import { UserProfileStore } from "@/store/useUserProfileStore";
import { spyUseParams } from "@/testUtils/navigation";
import { flushPromise, render } from "@/testUtils/testUtils";
import AsrService, { AsrServiceParams } from ".";

jest.mock("@/store/usePageLoadingStore", () => {
  return { ...jest.requireActual("@/store/usePageLoadingStore"), __esModule: true };
});

jest.mock("@/store/useUserProfileStore", () => {
  return { ...jest.requireActual("@/store/useUserProfileStore"), __esModule: true };
});

describe("AsrService", () => {
  let mockPageLoadingStore: PageLoadingStore;
  let mockUserProfileStore: UserProfileStore;
  let params: AsrServiceParams;

  beforeEach(() => {
    params = { service: "order-asr" };
    spyUseParams(params);

    mockPageLoadingStore = {
      isPageLoading: false,
      setPageLoading: jest.fn(),
    };

    mockUserProfileStore = {
      data: { corporate: "corporate" },
    } as UserProfileStore;

    jest.spyOn(UsePageLoadingStore, "default").mockImplementation(() => mockPageLoadingStore);
    jest.spyOn(UseUserProfileStore, "default").mockImplementation(() => mockUserProfileStore);

    jest.spyOn(RemoteConfig, "getValue").mockImplementation((_, key): any => {
      if (key === "DashboardMenu") {
        return { asString: () => JSON.stringify(mockDashboardMenuConfigResult) };
      }
    });
  });

  const renderAsrService = async () => {
    const view = render(<AsrService />);
    await flushPromise();
    return view;
  };

  it("should render correctly", async () => {
    const { asFragment } = await renderAsrService();
    expect(asFragment()).toMatchSnapshot();
  });

  const titleTests = [
    [ASR_SERVICE.ORDER_ASR, "Order ASR"],
    [ASR_SERVICE.SOAP_ASR, "SOAP ASR"],
  ];
  titleTests.forEach(([service, title]) => {
    it("should display title correctly", async () => {
      params.service = service;
      await renderAsrService();

      expect(document.title).toBe(title);
    });
  });
});
