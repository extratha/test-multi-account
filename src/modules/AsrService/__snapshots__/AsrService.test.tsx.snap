// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsrService should render correctly 1`] = `
<DocumentFragment>
  <title>
    Order ASR
  </title>
  .emotion-0 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 24px;
  overflow: hidden;
}

.emotion-1 {
  background-color: #fff;
  color: unset;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 16px;
  box-shadow: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
}

.emotion-2 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-3 {
  width: 100%;
  height: 100%;
  border: 0px;
}

<div
    class="MuiStack-root emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-1"
    >
      <div
        class="MuiStack-root emotion-2"
      >
        <iframe
          allow="camera; microphone"
          class="emotion-3"
          src="url://service-0?source=corporate"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
